test/kyberNetwork.js

```diff
@@ -2874,6 +2873,121 @@ contract('KyberNetwork', function(accounts) {
             let avgGas = cumulativeGas.div(numTrades);
             log("average gas usage " + numTrades + " buys. token to token: " + avgGas.floor().valueOf());
         });
+
+        it("should revert if token A -> ETH has no rate (resulting wei amount is zero), but has ETH -> token B rates", async function () {
+            let tokenSrcInd = 1;
+            let tokenDestInd = 2;
+            let tokenSrc = tokens[tokenSrcInd];
+            let tokenDest = tokens[tokenDestInd];
+            let srcAmount = 100;
+            let maxDestAmount = (new BigNumber(10)).pow(18);
+
+            //check that there is token A -> token B rate
+            rate = await network.getExpectedRate(tokenSrc.address, tokenDest.address, srcAmount);
+            assert(rate[0].valueOf() > 0, "token A -> token B rate is zero from the beginning");
+
+            //unlist ETH <> token A pair
+            let reservesToUnlistAndRelist = [];
+            reserveAddress = await network.reservesPerTokenSrc(tokenSrc.address, 0);
+            while (reserveAddress) {
+              reservesToUnlistAndRelist.push(reserveAddress);
+              await network.listPairForReserve(reserveAddress, tokenSrc.address, true, true, false, {from: operator});
+              try {
+                reserveAddress = await network.reservesPerTokenSrc(tokenSrc.address, 0);
+              } catch (e) {
+                break;
+              }
+            }
+
+            //check that there token A -> ETH rate is zero
+            rate = await network.getExpectedRate(tokenSrc.address, ethAddress, srcAmount);
+            assert.equal(rate[0].valueOf(), 0, "token A -> ETH rate not zero");
+            //check that there is ETH -> token B rate
+            rate = await network.getExpectedRate(ethAddress, tokenDest.address, srcAmount);
+            assert(rate[0].valueOf() > 0, "ETH -> token B rate is zero");
+
+            //token A -> token B rate should be zero
+            rate = await network.getExpectedRate(tokenSrc.address, tokenDest.address, srcAmount);
```

Comment:
before your change this reverts?
I think it should revert only for specific reserves that don't handle zero quantity input..